src/Symfony/Component/String/ByteString.php

```diff
@@ -129,27 +129,23 @@ public function chunk(int $length = 1): array
         return $chunks;
     }
 
-    public function endsWith($suffix): bool
+    public function endsWith(string|iterable|AbstractString $suffix): bool
     {
-        if ($suffix instanceof parent) {
+        if ($suffix instanceof AbstractString) {
             $suffix = $suffix->string;
-        } elseif (\is_array($suffix) || $suffix instanceof \Traversable) {
+        } elseif (!\is_string($suffix)) {
             return parent::endsWith($suffix);
-        } else {
-            $suffix = (string) $suffix;
```

Comment:
This change removes support for `Stringable` objects, doesn't it?

Reply:
See the added type hint: no Stringable object can get through it.
Ppl that use strict types might be affected, but strict types aren't covered by our BC promise.